sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -188,24 +208,134 @@ private AnalyzeSentimentResult convertToAnalyzeSentimentResult(DocumentSentiment
+                    opinionSentiments.add(toOpinionSentiment(
+                        findSentimentOpinion(opinionPointer, documentSentimentList)));
+                }
+            });
+
+            minedOpinions.add(new MinedOpinions(
+                new AspectSentiment(sentenceAspect.getText(),
+                    TextSentiment.fromString(sentenceAspect.getSentiment().toString()),
+                    sentenceAspect.getOffset(), sentenceAspect.getLength(),
+                    toSentimentConfidenceScores(sentenceAspect.getConfidenceScores())),
+                new IterableStream<>(opinionSentiments)));
+        });
+
+        return new IterableStream<>(minedOpinions);
+    }
+
+    /*
+     * Transform type AspectConfidenceScoreLabel to SentimentConfidenceScores.
+     */
+    private SentimentConfidenceScores toSentimentConfidenceScores(
+        AspectConfidenceScoreLabel aspectConfidenceScoreLabel) {
+        return new SentimentConfidenceScores(aspectConfidenceScoreLabel.getNegative(), NEUTRAL_SCORE_ZERO,
+            aspectConfidenceScoreLabel.getPositive());
+    }
+
+    /*
+     * Transform type SentenceOpinion to OpinionSentiment.
+     */
+    private OpinionSentiment toOpinionSentiment(SentenceOpinion sentenceOpinion) {
+        return new OpinionSentiment(sentenceOpinion.getText(),
+            TextSentiment.fromString(sentenceOpinion.getSentiment().toString()),
+            sentenceOpinion.getOffset(), sentenceOpinion.getLength(), sentenceOpinion.isNegated(),
+            toSentimentConfidenceScores(sentenceOpinion.getConfidenceScores()));
+    }
+
+    /*
+     * Parses the reference pointer to an index array that contains document, sentence, and opinion indexes.
+     */
+    private int[] parseRefPointerToIndexArray(String referencePointer) {
+        // The pattern always start with character '#', the opinion index will existing in specified sentence, which
+        // is under specified document.
+        // example: #/documents/0/sentences/0/opinions/0
+        final String patternRegex = "#\\/documents\\/(\\d+)\\/sentences\\/(\\d+)\\/opinions\\/(\\d+)";
+        final Pattern pattern = Pattern.compile(patternRegex);
+        final Matcher matcher = pattern.matcher(referencePointer);
+        final boolean isMatched = matcher.find();
+
+        // The first index represents the document index, second one represents the sentence index,
+        // third ond represents the opinion index.
+        final int[] result = new int[3];
+        if (isMatched) {
+            String[] segments = referencePointer.split("/");
+            result[0] = Integer.parseInt(segments[2]);
+            result[1] = Integer.parseInt(segments[4]);
+            result[2] = Integer.parseInt(segments[6]);
+        } else {
+            throw logger.logExceptionAsError(new RuntimeException(
+                String.format("'referencePointer' %s is not a valid opinion pointer", referencePointer)));
```

Comment:
It should throw since it is validating on the service's response. But I will change the Exception type to IllegalStateException since RuntimeException is too generic.